morf-core/src/main/java/org/alfasoftware/morf/sql/element/NullFieldLiteral.java

```diff
@@ -15,28 +15,34 @@
 
 package org.alfasoftware.morf.sql.element;
 
+import org.alfasoftware.morf.metadata.DataType;
 import org.alfasoftware.morf.sql.Statement;
 import org.alfasoftware.morf.util.DeepCopyTransformation;
-import org.apache.commons.lang.StringUtils;
 
 /**
  * Provides a representation of a null literal field value to be used in a {@link Statement}.
  *
  * @author Copyright (c) Alfa Financial Software 2010
+ * @deprecated Use {@link FieldLiteral#nullLiteral()}.
  */
+@Deprecated
 public class NullFieldLiteral extends FieldLiteral {
 
+  public NullFieldLiteral() {
+    super();
+  }
+
+
+  NullFieldLiteral(String alias) {
+    super(alias, null, DataType.NULL);
+  }
+
   /**
    * @see org.alfasoftware.morf.sql.element.AliasedField#deepCopyInternal(DeepCopyTransformation)
    */
   @Override
-  protected AliasedField deepCopyInternal(DeepCopyTransformation transformer) {
-    return new NullFieldLiteral();
+  protected NullFieldLiteral deepCopyInternal(final DeepCopyTransformation transformer) {
+    return new NullFieldLiteral(this.getAlias());
   }
 
-
-  @Override
-  public String toString() {

```

Comment:
Why has the toString gone?

Reply:
because the ancestor does it for us?

Reply:
Correct, and in a better way ("NULL" instead of "null" which was a bit misleading).  You've reminded me though that I need to tweak the dialect so a null FieldLiteral is equivalent to a NullFieldLiteral.  This was in this PR and got culled by accident when I was trying to slim it down to a manageable size.

Reply:
Could you elevate this to an review issue?

Reply:
Fixed